cmake/StandardProjectSettings.cmake

```diff
@@ -0,0 +1,370 @@
+include(GNUInstallDirs) # Standard install dirs
+
+# Uniform how we define BUILD_STATIC or BUILD_SHARED_LIBS (common practice)
+if(DEFINED BUILD_STATIC)
+    if(DEFINED BUILD_SHARED_LIBS)
+        if(${BUILD_SHARED_LIBS} AND ${BUILD_STATIC})
+            message(FATAL_ERROR "Conflicting arguments for BUILD_SHARED_LIBS=${BUILD_SHARED_LIBS} and BUILD_STATIC=${BUILD_STATIC}")
+        endif()
+    else()
+        set(BUILD_SHARED_LIBS NOT ${BUILD_STATIC})
+    endif()
+else()
+    if(NOT DEFINED BUILD_SHARED_LIBS)
+        set(BUILD_SHARED_LIBS ON)
```

Comment:
My personal preference is to use the `set(<variable> <value>... CACHE <type> <docstring> [FORCE])` over `option` everywhere. Since options are just  simple on/off but sets can serve for everything. This will keep it all uniform.

But this specific code block was added to handle our custom BUILD_STATIC option in multiple repo's. I'm more inclined to remove the BUILD_STATIC flag since the `BUILD_SHARED_LIBS` is a global CMake variable https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html?highlight=build_shared_libs

Reply:
Yeah, getting along with `BUILD_SHARED_LIBS` would indeed be better then.